xla/backends/gpu/collectives/gpu_clique_key.h

```diff
@@ -68,10 +68,17 @@ class GpuCliqueKey : public CliqueKey {
 
   CollectiveStreamId stream_id() const;
 
+  // Device generating the unique id for this key
+  GlobalDeviceId root_device() const;
+
   // Returns true if this clique is a subset of `other`: both cliques have the
   // same `stream_id` and all clique devices are part of `other` clique.
   bool IsSubsetOf(const CliqueKey& other) const final;
 
+  // Returns a copy of the key (subkey) with the root device properly set given
+  // nroots and root_seq_id. The subkey is used to generate a NcclCliqueId.
```

Comment:
+ @frgossen I feel that a bit more details would be helpful, I don't have the context, but the prorblem is that after reading this documentation I still don't have any idea what's going on :)